### PR TITLE
feat: event creation validation, relationships

### PR DESCRIPTION
**Pull request by stevenobird**
This PR is a draft for validation on Event form requests.

To be able to create events, you need to have the `create-events` permission on your current user.  
To achieve that via command line, you can do the following in the artisan tinker CLI:

Open Artisan Tinker:   
```shell
$ sail artisan tinker
```

retrieve your user by email
```shell
> $user = App\Models\User::whereEmail('my-mail@example.org')->first();
```

... or by the id:
```shell
> $user = App\Models\User::find(1)
```

then, assign the organizer role to your user:
```shell
> $user->assignRole('Veranstalter')
```

that should make the `authorize()` method in `app/Http/Requests/StoreEventRequest` to return true and therefore enable you to create events.

I've also added a redirect to `events/{event-slug}`, but it will give you an empty page, since there's no event detail page yet.

### CODE REVIEW

**Comment by Ryuno-Ki** [comment]
```
> $user->assignRole('Veranstalter');

   Spatie\Permission\Exceptions\RoleDoesNotExist  There is no role named `Veranstalter` for guard `web`.
```

I have migrated everything but `sail artisan db:seed` fails with

<details>
<summary>an error</summary>

<pre>
 INFO  Seeding database.  


   Illuminate\Database\UniqueConstraintViolationException 

  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'test@example.com' for key 'users_email_unique' (Connection: mariadb, SQL: insert into `users` (`name`, `email`, `email_verified_at`, `password`, `two_factor_secret`, `two_factor_recovery_codes`, `remember_token`, `profile_photo_path`, `current_team_id`, `updated_at`, `created_at`) values (Test User, test@example.com, 2024-10-15 12:47:54, $2y$12$8C.LX1c2Vj9QC2jUlPT61ucXjkuAgjZCOazYkNCxptfJudflgKapu, ?, ?, JWVbJWH6Rn, ?, ?, 2024-10-15 12:47:54, 2024-10-15 12:47:54))

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:820
    816▕         // message to include the bindings with SQL, which will make this exception a
    817▕         // lot more helpful to the developer instead of just the database's errors.
    818▕         catch (Exception $e) {
    819▕             if ($this->isUniqueConstraintError($e)) {
  ➜ 820▕                 throw new UniqueConstraintViolationException(
    821▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    822▕                 );
    823▕             }
    824▕ 

      +16 vendor frames 

  17  database/seeders/DatabaseSeeder.php:18
      Illuminate\Database\Eloquent\Factories\Factory::create()
      +23 vendor frames 

  41  artisan:13
      Illuminate\Foundation\Application::handleCommand()
</pre>
</details>

Deleting the row and re-running the command succeeds.

However, I have no data in roles, permissions or organizers table. Have you seeded them on your end?

File: app/Http/Requests/StoreEventRequest.php [hunk 1]
```diff
@@ -11,7 +11,7 @@ class StoreEventRequest extends FormRequest
      */
     public function authorize(): bool
     {
-        return true;
+        return auth()->user()->can('create-event');
```
Comment (Ryuno-Ki) [inline app/Http/Requests/StoreEventRequest.php:14]: Where is this imported from?
I have an `undefined method user()` on my end.

Comment (Ryuno-Ki) [reply app/Http/Requests/StoreEventRequest.php:14]: Wait, let me guess: Event -> Organizer -> User?

File: app/Http/Controllers/EventController.php [hunk 2]
```diff
@@ -24,27 +23,24 @@ public function index()
...
+        $event = Event::create([
+            'title' => $request->title,
+            'description' => $request->description,
+            'starts_at' => $request->starts_at,
+            'ends_at' => $request->ends_at,
         ]);
 
-        return redirect()->route('events.index');
+        return redirect()->route('events.show', [
+            'event' => $event->get,
```
Comment (Ryuno-Ki) [inline app/Http/Controllers/EventController.php:42]: Das `->get` hat bei mir einen Fehler geschmissen.

Comment (stevenobird) [reply app/Http/Controllers/EventController.php:42]: War wohl ein Überbleibsel. Nur $event sollte reichen.

**Comment by stevenobird** [comment]
> ```
> > $user->assignRole('Veranstalter');
> 
>    Spatie\Permission\Exceptions\RoleDoesNotExist  There is no role named `Veranstalter` for guard `web`.
> ```
> 
> I have migrated everything but `sail artisan db:seed` fails with
> 
> an error
> Deleting the row and re-running the command succeeds.
> 
> However, I have no data in roles, permissions or organizers table. Have you seeded them on your end?

Sorry, forgot to mention that you need seeding the permissions and roles:

```shell
$ sail artisan db:seed --class=RoleAndPermissionSeeder
```

**Pull request closed without merging**